internal/app/go-filecoin/connectors/retrieval_market/client.go

```diff
@@ -139,12 +139,12 @@ func (r *RetrievalClientConnector) getBlockHeight(tok shared.TipSetToken) (abi.C
 }
 
 func (r *RetrievalClientConnector) getBalance(ctx context.Context, account address.Address, tok shared.TipSetToken) (types.AttoFIL, error) {
-	ts, err := r.getTipSet(tok)
-	if err != nil {
+	var tsk block.TipSetKey
+	if err := tsk.UnmarshalCBOR(tok); err != nil {
```

Comment:
This will work but its idiomatic to use `encoding.Decode(tok, &tsk)` assuming tok is a byte slice underneath the hood

Reply:
This makes sense. I've been given this direction before; spaced out and used the wrong method this time. Will update.